flytekit/types/file/file.py

```diff
@@ -489,7 +489,7 @@ async def async_to_literal(
         meta = BlobMetadata(type=self._blob_type(format=FlyteFilePathTransformer.get_format(python_type)))
 
         if isinstance(python_val, FlyteFile):
-            source_path = python_val.path
+            source_path = str(python_val.path)
             self.validate_file_type(python_type, source_path)
```

Comment:
1. can you add a unit test for this case?
2. does flytedirectory have the same error?

Reply:
Hi @Future-Outlier,

1. I've created a new unit test for this specific case.
2. As far as I know, flytedirectory can well handle the reported error. I can add one more test to demonstrate.

Reply:
Yes please do this, in the future, we will change the implementation of flyte file and flyte director, this can help prevent error

Reply:
I've added one more test on `FlyteDirectory`, which also passes!